app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -374,8 +374,8 @@ def self.default_blacklisted_event_names
       )
     end
 
-    def build_metric_id(type, resource, metric_id)
-      "#{type}I~R~[#{resource[:middleware_server][:feed]}/#{resource[:nativeid]}]~#{type}T~#{metric_id}"
+    def build_metric_id(type, feed_id, resource_id, metric_id)
```

Comment:
One side comment.
I guess this function is only used in the availability context.
Expression "#{type}I~R~[#{feed_id}/#{resource_id}]~#{type}T~#{metric_id}" is not valid for all types of metrics so I suggest to just rename to build_availability_id / build_availabiliby_metric_id to mark this use case.
In other metrics the idea is to operate at metric object to not deal with id manipulation.
But I guess for availability it's a different use case, so I guess it would be clear (for future maintainers to do this minor rename).